### ISSUE TITLE
Initializing a child struct inside a generic struct fails

### ISSUE THREAD

**Issue by clindholm** [vlang/v#18040]
### Describe the bug

I see no reason why this shouldn't work. Making `child` mutable and setting after initialization does work.

```
src/main.v:11:9: error: generic struct init must specify type parameter, e.g. Foo[T]
     9 |
   10 | fn new[T](items []T) Test[T] {
   11 |     return Test[T]{
        |            ~~~~~~~~
   12 |         child: Child{}
   13 |     }
```

### Expected Behavior

Should compile and run

### Current Behavior

Fails

### Reproduction Steps

```
module main

struct Child {
}

struct Test[T] {
	child Child
}

fn new[T]() Test[T] {
	return Test[T]{
		child: Child{}
	}
}

fn main() {
	t := new[int]()
	println(t)
}
```

### V version

V 0.3.3 f598bbd

### Environment details (OS name and version, etc.)

V full version: V 0.3.3 f598bbd
OS: linux, Linux version 6.1.25-1-lts (linux-lts@archlinux) (gcc (GCC) 12.2.1 20230201, GNU ld (GNU Binutils) 2.40) #1 SMP PREEMPT_DYNAMIC Thu, 20 Apr 2023 14:01:39 +0000
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-8109U CPU @ 3.00GHz

vexe mtime: 2023-04-24 11:06:40

VTMP: OK, value: /tmp/v_1000

Git version: git version 2.40.0
Git vroot status: weekly.2023.15-57-gf598bbde
.git/config present: true

CC version: cc (GCC) 12.2.1 20230201
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

**Comment by clindholm** [comment]
Also this works, in case someone else needs a quick fix:

```
module main

struct Child {
	val int
}

struct Test[T] {
	child Child = Child{1}
}

fn new[T]() Test[T] {
	return Test[T]{}
}

fn main() {
	t := new[int]()
	println(t)
}
```

**Comment by DevrosTheOne** [comment]
You can have an empty Child structure assigned as long as you don't do it explicitly.  Created by default.  

```v
struct Child {
}

struct Test[T] {
	child Child
}

fn new[T]() Test[T] {
	return Test[T]{
		// child: Child{}
	}
}

fn main() {
	t := new[int]()
	println(t)
}
```
```
Test[int]{
    child: Child{}
}
```

**Comment by clindholm** [comment]
@DevrosTheOne 
Thanks, great tip!
But this was just the simplest code I could write that would trip up the checker. In the code where I actually found this, the child was not default.

Another way to work around this, that I found, is to use a temporary variable:
```
fn new[T]() Test[T] {
	child := Child{2}
	return Test[T]{
		child: child
	}
}
```
which helps for cases like this, that also fail:
```
fn new[T](i int) Test[T] {
	return Test[T]{
		child: Child{i}
	}
}
```

So the bug, all in all, is not that big of a deal with the workarounds, but it is unexpected.
